Mono.Debugger.Soft/Mono.Debugger.Soft/VirtualMachineManager.cs

```diff
@@ -111,20 +111,19 @@ public static IAsyncResult BeginLaunch (ProcessStartInfo info, AsyncCallback cal
 				socket.Close ();
 			};
 
-			LaunchCallback c = new LaunchCallback (LaunchInternal);
-			return c.BeginInvoke (p, info, socket, callback, socket);
+			Task<VirtualMachine> t3 = Task<VirtualMachine>.Run (() => {
```

Comment:
Doing a `Task.Run` here does not mirror previous functionality. Maybe simplify it to:

```csharp
return LaunchInternal (p, info, socket)
    .ContinueWith (antecendent => callback (antecendent));
```